### PR TITLE
Kata/collections/checkpoint/exercise

### PR DESCRIPTION
**Pull request by vikrantdheer**

### CODE REVIEW

File: src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/tasks/CheckIn.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public CheckIn(Pet pet, PetHotel petHotel) {
+        this.pet = pet;
+        this.petHotel = petHotel;
+    }
+
+    @Step("{0} checks-in #pet into #petHotel")
+    @Override
+    public <T extends Actor> void performAs(T actor) {
+        BookingResponse bookingResponse = petHotel.checkIn(pet);
```
Comment (wakaleo) [inline src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/tasks/CheckIn.java]: Here, you never use the 'bookingResponse' variable, so you never actually check the response coming back from the hotel.

File: src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/questions/TheBookingStatus.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import net.serenitybdd.screenplay.Question;
+import serenitylabs.tutorials.vetclinic.Pet;
+import serenitylabs.tutorials.vetclinic.collections.katas.BookingResponse;
+
+/**
+ * Created by vdheer on 9/20/2016.
+ */
+public class TheBookingStatus implements Question<Boolean> {
+
+    private BookingResponse bookingResponse = new BookingResponse(true);
```
Comment (wakaleo) [inline src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/questions/TheBookingStatus.java]: Here, BookingResponse always created with a parameter of 'true', so the question will always return true.

File: src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/questions/TheBookingStatus.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    private Pet pet;
+
+    public TheBookingStatus(Pet pet) {
+        this.pet = pet;
```
Comment (wakaleo) [inline src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/questions/TheBookingStatus.java]: The pet is never actually used. Remember, the booking status isn't directly related to the pet, but is the outcome the booking process when you try to book a pet into a hotel.

File: src/test/java/serenitylabs/tutorials/vetclinic/collections/katas/WhenBookingPetsIntoAPetHotel.java [hunk 4]
```diff
@@ -1,54 +1,186 @@
...
+        PetHotel petHotel = APetHotel.with(19).checkedIn();
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet goldy = Pet.fish().named("Goldy");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel), CheckIn.aPet(goldy).in(petHotel));
+
+        //THEN
+        then(vikrant).should(seeThat(TheWaitingStatus.of(goldy).isConfirmed(), is(false)));
+    }
```
Comment (wakaleo) [inline src/test/java/serenitylabs/tutorials/vetclinic/collections/katas/WhenBookingPetsIntoAPetHotel.java]: This test is missing an edge case. If the hotel is full, the booking status should fail, e.g.
```
    @Test
    public void should_notify_owner_that_the_hotel_is_full() throws Exception {
        //GIVEN
        PetHotel petHotel = APetHotel.with(19).checkedIn();
        Pet tommy = Pet.dog().named("Tommy");
        Pet goldy = Pet.fish().named("Goldy");

        //WHEN
        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel), CheckIn.aPet(goldy).in(petHotel));

        //THEN
        then(vikrant).should(seeThat(TheBookingStatus.of(goldy).isConfirmed(), is(false)));
        and(vikrant).should(seeThat(TheWaitingStatus.of(goldy).isConfirmed(), is(false)));
    }
```

File: src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/PetHotel.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    private static String hotelName;
+
+    public List<Pet> getRegisteredPets() {
+        return new ArrayList<>(registeredPets);
+    }
+
+    public BookingResponse checkIn(Pet pet) {
+        return (registeredPets.size() < DEFAULT_MAXIMUM_CAPACITY)
+                ? new BookingResponse(registeredPets.add(pet))
+                : new BookingResponse(waitingListOfPets.add(pet));
```
Comment (wakaleo) [inline src/main/java/serenitylabs/tutorials/vetclinic/collections/katas/PetHotel.java]: This will return a positive booking response in both cases, so there is no way for the customer to know if the pet was successfully booked or not.
